src/MonitorProfiler/MainProfiler/ExceptionTracker.cpp

```diff
@@ -0,0 +1,392 @@
+        - ExceptionSearchFilterEnter: Invoked when beginning to evaluate the filter.
+        - ExceptionSearchFilterLeave: Invoked when finishing the evaluation of the filter.
+      - ExceptionSearchCatcherFound: Invoked only if a matching filter is found.
+      - ExceptionSearchFunctionLeave: Invoked when finishing the inspection of the function for a matching handler.
+      If a matching handler was found, the exception catcher finding phase ends after the ExceptionSearchFunctionLeave callback.
+    - Phase 3: Unwinding.
+      The following are invoked for each frame starting at the leaf frame (the frame from which the exception is thrown):
+      - ExceptionUnwindFunctionEnter: Invoked when beginning to unwind a frame.
+      If a matching handler was found, the unwind phase ends after the invocation of
+      the ExceptionUnwindFunctionEnter callback for the frame that contains the matching handler.
+      - ExceptionUnwindFunctionLeave: Invoked when finishing the unwind of a frame.
+      If all of the frames are unwound to completion, the process is terminated.
+
+    Unhandled Exception Detection Algorithm
+
+    A subset of the aboved described callbacks is used to determine if an unhandled exception has occurred:
+    - ExceptionThrown: Record the originating FunctionID and the existance of the exception.
+    - ExceptionSearchCatcherFound: Record that the exception will be handled in a catching FunctionID.
+    - ExceptionUnwindFunctionEnter: If the exception was not handled (ExceptionSearchCatcherFound was not invoked),
+      then the exception is unhandled at this point. Do some extra bookkeeping in the case when the exception is
+      handled so that the exception is cleared once the frame with the corresponding catching FunctionID is encountered.
+
+    Current pitfalls:
+    - Algorithm doesn't account for exceptions thrown within exception filters. These will be seen as a new set of
+      callback invocations starting with throwing, searching, and unwinding, however, the callstack does not unwind
+      out of the exception filter and the original exception processing is resumed.
+    - Algorithm doesn't account for exceptions thrown within finally blocks. Presumably, these will be seen as new set
+      of callback invocations starting with throwing, searching, and unwinding. The original exception processing is
+      superceded by the new set of callbacks for the new exception.
+*/
+
+
+ExceptionTracker::ExceptionTracker(
+    const shared_ptr<ILogger>& logger,
+    const shared_ptr<ThreadDataManager> threadDataManager,
+    ICorProfilerInfo2* corProfilerInfo)
+{
+    _corProfilerInfo = corProfilerInfo;
+    _logger = logger;
+    _threadDataManager = threadDataManager;
+}
+
+void ExceptionTracker::AddProfilerEventMask(DWORD& eventsLow)
+{
+    eventsLow |= COR_PRF_MONITOR::COR_PRF_ENABLE_STACK_SNAPSHOT;
```

Comment:
@wiktork, we only use this when the logging level is Debug or lower. Do you think there is any benefit to dynamically adding this when we only need it? This might be moot if the stacks feature is using it unconditionally.

Reply:
I'm not aware of any significant perf impact from this flag, but I think as we add more features here we may end up with configuration that enables or disables features based on perf impact. Only adding the flags you actually need is probably the right approach.